### ISSUE TITLE
Missing Ammy.Syntax project files

### ISSUE THREAD

**Issue by isukces** [AmmyUI/AmmyUI#148]
Seems that git repository doesn't contains following files for `Ammy.Syntax` project
`Nemerle.Linq`
`Nitra.Compiler`
Are they necessary or just remove references from csproj?

**Comment by ionoy** [comment]
did you run `build-ammy.cmd`?

**Comment by isukces** [comment]
Yes. MSbuild from VS2019 was used.  I got compilation error
```
 "C:\programs\github\AmmyUI\Ammy.sln" (default target) (1) ->
       "C:\programs\github\AmmyUI\src\Core\Ammy.Syntax\Ammy.Syntax.nproj" (default target) (5) ->
       (CoreCompile target) ->
         C:\Program Files (x86)\Nemerle\net-4.0\Nemerle.MSBuild.targets(289,5): error : internal compiler error: got some unknown exception of type System.TypeLoadException: Could not load type 'Nemerle.Collections.ListEnumerator`1' fro
       m assembly 'Nitra.Grammar, Version=0.1.4015.0, Culture=neutral, PublicKeyToken=e080a9c724e2bfcd' due to value type mismatch. [C:\programs\github\AmmyUI\src\Core\Ammy.Syntax\Ammy.Syntax.nproj]
```

I know Nemerle is not supported with VS2019, so did you use VS2017 or make some trick?

**Comment by ionoy** [comment]
I do have VS2017 installed. Can you try installing it also?

**Comment by isukces** [comment]
Still nothing. I found that file `run-msbuild.bat` works better if VS version is specified, so
```
setlocal enabledelayedexpansion

for /f "usebackq tokens=*" %%i in (`vswhere -version "[15.0,16.0)" -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe`) do (
  "%%i" %*
)
```
It chooses VS 2019 when `-latest` parameter is used. 

MSBuild and compiler loads Nemerle and Nitra related dlls both from `C:\Program Files (x86)\Nemerle\net-4.0` and local `lib` folder. I synchronized these folders so they contain the same dll versions but it doesn't help.

Seems that preparing a building environment is not a trivial task 😒 especially when Nemerle is no longer supported. 
I think there is no sense in fighting this problem. Builing Ammy by myself is not critical for me.

**Comment by ionoy** [comment]
Ammy now uses local Nemerle binaries so you definitely need to uninstall Nemerle for this build to work. Sorry, I forgot to mention that.

Btw, can you use the latest NuGet package that I have uploaded?

**Comment by isukces** [comment]
I use the latest NuGet version in my projects.
I've successfully compiled `Ammy.Backend.dll` when I moved files from the `lib` folder to `C:\Program Files (x86)\Nemerle\net-4.0`, rid off `lib` folder and changed all references to `C:\Program Files (x86)\Nemerle\net-4.0`.

I also found that `Microsoft.Cci.MetadataModel.dll` is twice, and only version 1.0.65055.0 works properly.

Unfortunately, not the whole solution was compiled that way. It stopped -as I remember - on `Ammy.Syntax` project or something.

Seems that uninstalling Nemerle not helps because of the missing `Nemerle.MSBuild.targets` file. 
I'm still trying...

**Comment by ionoy** [comment]
`Nemerle.MSBuild.targets` should be resolved from `lib\Nemerle-bin\net-4.0\` if you don't have Nemerle installed. You need to use the specific Nemerle version from the repo, that's why removing Nemerle installation is important. 

I suggest you 
1) Uninstall Nemerle
2) Delete current `AmmyUI` repo
3) Clone again
4) Run `build-ammy.cmd`

I think that should work.

**Comment by isukces** [comment]
After uninstalling Nemerle i had to remove manually `NemerleBinPathRoot` environment variable. 
Everything works properly now. 
I kept my `run-msbuild.bat` modification.
Thanks.
